AppKit/CPViewController.j

```diff
@@ -176,6 +176,58 @@ var CPViewControllerCachedCibs;
         _view = [CPView new];
 }
 
+- (void)loadViewAsynchronously:(Function/*(view, error)*/)aHandler
```

Comment:
Semantic is everything and defines a framework. Maybe this comment is not the best place to widen the discussion, but since it is a name issue here, I'll try to explain my view. Also it is something that I am working with right now so I'll be happy to have some discussion around it.

We are using in Cappuccino a rule - that we should mimic Cocoa functionality and name standard as long as it is possible. The rule of thumb is to make the API's clear, distinct and obvious, but not to overload the semantic. And I think NeXT and Apple has excelled in this in the NS-frameworks.

But the two frameworks differs in some very basic ways and therefor we have to do some exceptions once in a while. 
In this special case the method is loading a view asynchronously and that is one of the times blocks excels in an method API. The NSViewController's `loadView` the method is not asynchronous since, and here the two framework differs. In Cocoa/C/ObjC we can use threads but since JavaScript is single-threaded, it uses asynchronous technics to achieve throughput in the execution flow.
The NSViewController's `loadView` is synchronous and does not have to have a callback but in Cappuccino we have to make it asynchronous and need a callback. If we look at the cases in Cocoa where blocks is used there is when the block is called inside the method and  when it is called after as a completion handler. (The word block is used with care since it has more than one meaning aka. also 'to block' something.)

An example of the first case is in, for instance, NSArray's `enumerateObjectsUsingBlock:` and the second case is in asynchronous cases like NSURLSession's `downloadTaskWithURL:completionHandler:`. The 'loadView' method is more of the second case and if using cocoa name standard it maybe would be nice to call it `loadViewWithCompletionHandler:`. This is clear and distinct.
To explicitly point out that it is an asynchronous call is in my opinion to overload the method name. This may advantageously be pointed out in the documentation. My view here is based on the basic differences in the two frameworks and since JavaScript relies on asynchronicity it is redundant. 

If we look forward in the Cappuccino framework's future we will soon see that node.js heavily relies on asynchronicity in it's API, here the normal case is not synchronous but asynchronous. An example is the module FileSystem where functions like `fs.exists`, `fs.link`, `fs.open`, `fs.chmod` etc. are asynchronous and the normal case, but there are other functions like `fs.existsSync`, `fs.linkSync`, `fs.openSync` and `fs.chmodSync` if a synchronous call is explicitly preferred. 

I am sorry to overload this thread with the following and it should be continued elsewhere but it shed some lights on the problem but...
I am now implementing some more classes to Foundation to be used under node.js like CPFileManager and others and there I have to extend the NS API with asynchronous methods that are equivalent, like the method: 
`- (BOOL)moveItemAtURL:(CPURL)srcURL toURL:(CPURL)dstURL error:(CPError)error`
who will have a equivalent that may be called
`- (BOOL)moveItemAtURL:(CPURL)srcURL toURL:(CPURL)dstURL errorHandler:(Function)anErrorHandler completionHandler:(Function)aCompletionHandler`
The first method uses delegation and the other function blocks to handle errors and completion.

Seeing forward to continue this discussion.